common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/MicrosoftAuthorizationRequest.java

```diff
@@ -174,8 +182,16 @@ protected MicrosoftAuthorizationRequest(@SuppressWarnings(WarningType.rawtype_wa
         mDiagnosticOS = Device.getOsForEsts();
         mDiagnosticDM = Device.getModel();
         mDiagnosticCPU = Device.getCpu();
-        mDiagnosticMN = Device.getManufacturer();
-        mWorkProfileAvailable = Device.isInPersonalProfileButClouddpcWorkProfileAvailable();
+
+
+        // If the flight is enabled, set the fields
+        if (CommonFlightsManager.INSTANCE.getFlightsProvider().isFlightEnabled(CommonFlight.ENABLE_AM_API_WORKPROFILE_EXTRA_QUERY_PARAMETERS)) {
+            mDiagnosticMN = Device.getManufacturer();
+            mWorkProfileAvailable = Device.isInPersonalProfileButClouddpcWorkProfileAvailable();
+        } else {
+            mDiagnosticMN = null;
+            mWorkProfileAvailable = null;
+        }
```

Comment:
Do we need the else block? The default is null anyway, right?

Reply:
If they are final variables, then the else block is needed, getting an error about them not being initialized